src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/adjudications/AdjudicationsService.kt

```diff
@@ -364,10 +386,112 @@ class AdjudicationsService(
+    val prisonerNumber: String = punishmentEvent.additionalInformation.prisonerNumber
+    val telemetryMap = mutableMapOf(
+      "chargeNumber" to chargeNumber,
+      "prisonId" to prisonId,
+      "offenderNo" to prisonerNumber,
+    )
+
+    synchronise {
+      name = EntityType.PUNISHMENT.displayName
+      telemetryClient = this@AdjudicationsService.telemetryClient
+      retryQueueService = adjudicationRetryQueueService
+      eventTelemetry = telemetryMap
+
+      transform {
+        val mapping =
+          adjudicationMappingService.getMappingGivenChargeNumber(chargeNumber)
+            .also {
+              telemetryMap["adjudicationNumber"] = it.adjudicationNumber.toString()
+              telemetryMap["chargeSequence"] = it.chargeSequence.toString()
+            }
+
+        val adjudicationNumber = mapping.adjudicationNumber
+        val chargeSequence = mapping.chargeSequence
+
+        val punishments = adjudicationsApiService.getCharge(
+          chargeNumber,
+          prisonId,
+        ).reportedAdjudication.punishments
+
+        val nomisAwardsResponse =
+          nomisApiService.createAdjudicationAwards(adjudicationNumber, chargeSequence, CreateHearingResultAwardRequests(awardRequests = punishments.map { it.toNomisAward() }))
+            .also { telemetryMap["punishmentsCount"] = it.awardResponses.size.toString() }
+
+        AdjudicationPunishmentBatchMappingDto(
+          punishments = punishments.zip(nomisAwardsResponse.awardResponses) { punishment, awardResponse ->
+            AdjudicationPunishmentMappingDto(
+              dpsPunishmentId = punishment.id.toString(),
+              nomisBookingId = awardResponse.bookingId,
+              nomisSanctionSequence = awardResponse.sanctionSequence,
+            )
+          },
+        )
+      }
+      saveMapping { punishmentsMappingService.createMapping(it) }
+    }
+  }
+
   private inline fun <reified T> String.fromJson(): T =
     objectMapper.readValue(this)
 }
 
+private fun PunishmentDto.toNomisAward() = CreateHearingResultAwardRequest(
+  sanctionType = this.type.toNomisSanctionType(),
+  sanctionStatus = this.toNomisSanctionStatus(),
+  effectiveDate = this.schedule.startDate ?: this.schedule.suspendedUntil ?: LocalDate.now(),
+  sanctionDays = this.schedule.days,
+  compensationAmount = this.damagesOwedAmount?.toBigDecimal() ?: stoppagePercentage?.toBigDecimal(),
+  commentText = this.toComment(),
+  // TODO consecutive adjudication
+)
+
+private fun PunishmentDto.toComment(): String? =
+  // copy of existing logic from prison-api
+  when (this.type) {
+    PRIVILEGE ->
+      when (this.privilegeType) {
+        PunishmentDto.PrivilegeType.OTHER -> "Loss of ${this.otherPrivilege}"
+        else -> "Loss of ${this.privilegeType}"
+      }
+
+    DAMAGES_OWED -> "OTHER - Damages owed £${String.format("%.2f", this.damagesOwedAmount!!)}"
+    else -> null
+  }
+
+private fun PunishmentDto.toNomisSanctionStatus(): SanctionStatus {
```

Comment:
This is taken from prison-api but not sure it is correct - so need to validate and test better